pkg/pool-utils/contracts/BasePool.sol

```diff
@@ -592,17 +599,17 @@ abstract contract BasePool is
     /**
      * @dev Adds swap fee amount to `amount`, returning a higher value.
      */
-    function _addSwapFeeAmount(uint256 amount) internal view returns (uint256) {
+    function _addSwapFeeAmount(uint256 amount, bytes memory userData) internal view returns (uint256) {
         // This returns amount + fee amount, so we round up (favoring a higher fee amount).
-        return amount.divUp(getSwapFeePercentage().complement());
+        return amount.divUp(getSwapFeePercentage(userData, OperationType.SWAP).complement());
```

Comment:
This assumes the function will only be called in the context of a Swap operation, which might be true in the current codebase but is not necessary IIUC. So I wonder if it's not best to pass the OperationType as an argument too.